barretenberg/cpp/src/barretenberg/dsl/acir_format/mock_verifier_inputs.cpp

```diff
@@ -145,62 +196,255 @@ template <typename Flavor, class PublicInputs> HonkProof create_mock_pg_proof()
  */
 Goblin::MergeProof create_mock_merge_proof()
 {
-    using Flavor = MegaFlavor;
-    using FF = Flavor::FF;
-
-    std::vector<FF> proof;
+    Goblin::MergeProof proof;
     proof.reserve(MERGE_PROOF_SIZE);
 
-    FF mock_val(5);
-    auto mock_commitment = curve::BN254::AffineElement::one();
-    std::vector<FF> mock_commitment_frs = field_conversion::convert_to_bn254_frs(mock_commitment);
+    uint32_t mock_shift_size = 5; // Must be smaller than 32, otherwise pow raises an error
 
-    // Populate mock subtable size
-    proof.emplace_back(mock_val);
+    // Populate mock shift size
+    populate_field_elements<fr>(proof, 1, /*value=*/fr{ mock_shift_size });
 
     // There are 8 entities in the merge protocol (4 columns x 2 components: T_j, g_j(X) = X^{l-1} t_j(X))
     // and 8 evaluations (4 columns x 2 components: g_j(kappa), t_j(1/kappa))
     const size_t NUM_TRANSCRIPT_ENTITIES = 8;
     const size_t NUM_TRANSCRIPT_EVALUATIONS = 8;
 
     // Transcript poly commitments
-    for (size_t i = 0; i < NUM_TRANSCRIPT_ENTITIES; ++i) {
-        for (const FF& val : mock_commitment_frs) {
-            proof.emplace_back(val);
-        }
-    }
+    populate_field_elements_for_mock_commitments(proof, NUM_TRANSCRIPT_ENTITIES);
+
     // Transcript poly evaluations
-    for (size_t i = 0; i < NUM_TRANSCRIPT_EVALUATIONS; ++i) {
-        proof.emplace_back(mock_val);
-    }
+    populate_field_elements(proof, NUM_TRANSCRIPT_EVALUATIONS);
 
     // Shplonk proof: commitment to the quotient
-    for (const FF& val : mock_commitment_frs) {
-        proof.emplace_back(val);
-    }
+    populate_field_elements_for_mock_commitments(proof, 1);
 
     // KZG proof: commitment to W
-    for (const FF& val : mock_commitment_frs) {
-        proof.emplace_back(val);
-    }
+    populate_field_elements_for_mock_commitments(proof, 1);
 
     BB_ASSERT_EQ(proof.size(), MERGE_PROOF_SIZE);
 
     return proof;
 }
 
+/**
+ * @brief Create a mock pre-ipa proof which has the correct structure but is not necessarily valid
```

Comment:
might be nice to mention the eccvm in these comments/naming since pre-ipa isn't clear unless you're already in the eccvm context